src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProject.cs

```diff
@@ -246,31 +231,28 @@ private static PackageReference ToPackageReference(LibraryDependency library, Nu
 
         private async Task<PackageSpec> GetPackageSpecAsync()
         {
-            PackageSpec packageSpec = null;
-            await ThreadHelper.JoinableTaskFactory.RunAsync(async delegate
-            {
-                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                packageSpec = GetPackageSpec();
-            });
-
-            return packageSpec;
+            return await RunOnUIThread(GetPackageSpec);
         }
 
         private PackageSpec GetPackageSpec()
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
+            EnsureUIThread();
 
             var projectReferences = _project.GetLegacyCSProjProjectReferences(_desiredPackageReferenceMetadata)
                 .Select(ToProjectRestoreReference);
 
             var packageReferences = _project.GetLegacyCSProjPackageReferences(_desiredPackageReferenceMetadata)
                 .Select(ToPackageLibraryDependency);
 
+            var packageTargetFallback = _project.PackageTargetFallback?.Split(new[] { ';' })
+                .Select(NuGetFramework.Parse)
+                .ToList();
+
             var projectTfi = new TargetFrameworkInformation()
             {
                 FrameworkName = _project.TargetNuGetFramework,
-                Dependencies = packageReferences.ToList()
+                Dependencies = packageReferences.ToList(),
+                Imports = packageTargetFallback
```

Comment:
If imports exist you need to update the target framework, PackageSpec should handle this better, but there isn't a good way to do that today.

You can do the same thing as the restore target:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs#L226-L230